### PR TITLE
WT-12031 Add a new configuration item for background compaction to run once

### PR DESCRIPTION
**Pull request by etienneptl**
- Added a new configuration item `run_once` for the background compaction
- When configured with `run_once`, the background compaction will go to sleep after a full iteration

### CODE REVIEW

File: src/conn/conn_compact.c [hunk 1]
```diff
@@ -465,6 +465,14 @@ __background_compact_server(void *arg)
              * have been parsed.
              */
             if (uri->size == 0 || full_iteration) {
+                /* If the server was configured to run once, after a full iteration, we are done. */
+                if (full_iteration && conn->background_compact.run_once) {
```
Comment (etienneptl) [inline src/conn/conn_compact.c:469]: I am not a fan of that specific check, I wish there was a better way to change the background compaction state and update the relevant stat in one place. @Sean04, who do you think we could improve this?

**Comment by etienneptl** [comment]
There is one thing to confirm whether background compaction should resume from the last position or restart from the beginning when it is enabled with the `run_once` option.

**Comment by Sean04** [comment]
Regarding the questions:

1. Run once should always start from the beginning.
2. Run once shouldn't take the stats into consideration.

File: src/conn/conn_compact.c [hunk 2]
```diff
@@ -484,6 +496,10 @@ __background_compact_server(void *arg)
         __wt_spin_lock(session, &conn->background_compact.lock);
         running = conn->background_compact.running;
         if (conn->background_compact.signalled) {
+            /* If configured to run once, start from the beginning. */
+            if (running && conn->background_compact.run_once)
```
Comment (Sean04) [inline src/conn/conn_compact.c:500]: Is it guaranteed that we should be running at this point?

Comment (etienneptl) [reply src/conn/conn_compact.c:500]: Yes, you have to enable the background compaction server to set this setting.

Comment (etienneptl) [reply src/conn/conn_compact.c:500]: As discussed, remove the `running` flag and add an assert.

File: src/conn/conn_compact.c [hunk 3]
```diff
@@ -213,6 +213,10 @@ __background_compact_should_run(WT_SESSION_IMPL *session, const char *uri, int64
     if (compact_stat == NULL)
         return (true);
 
+    /* If we are running once, force compaction on the file. */
```
Comment (kommiharibabu) [inline src/conn/conn_compact.c:216]: Based on the `run_once` flag, we force each file to compact irrespective of its stats gathered earlier. I felt we shouldn't change this behaviour. I am fine with force, too, if others agree.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/conn/conn_compact.c (hunk 3 above)
Comment (etienneptl) [reply src/conn/conn_compact.c:257]: Updated the intent of that field in b9c413261c3966f8f656d4e7c218c6caf1869114. We want to force compaction on each table when `run_once` is set.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Comment by etienneptl** [comment]
evergreen merge

**Pull request merged**